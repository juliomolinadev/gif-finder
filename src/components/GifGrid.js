import React from "react";
import PropTypes from "prop-types";

const GifGrid = ({ word }) => {
	return (
		<div className="row row-cols-1 row-cols-sm-3 g-3">
			<div className="col">
				<div className="border rounded p-3">Card: {word}</div>
			</div>
			<div className="col">
				<div className="border rounded p-3">Card: {word}</div>
			</div>
			<div className="col">
				<div className="border rounded p-3">Card: {word}</div>
			</div>
		</div>
	);
};

GifGrid.propTypes = {
	word: PropTypes.string.isRequired
};

export default GifGrid;
