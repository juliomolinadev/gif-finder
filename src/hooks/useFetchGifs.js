import { useEffect, useState } from "react";
import { getGifs } from "../helpers/getGifs";

export const useFetchGifs = (word) => {
	const [state, setstate] = useState({ loading: true, gifs: [] });

	useEffect(() => {
		getGifs(word).then((gifs) => {
			setstate({ loading: false, gifs });
		});
	}, [word]);

	return state;
};
