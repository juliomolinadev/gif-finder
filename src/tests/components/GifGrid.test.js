import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";

import GifGrid from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe("Tests in <GifGrid />", () => {
	const word = "hi";

	test("Component should be displayed correctly", () => {
		useFetchGifs.mockReturnValue({
			loading: true,
			gifs: []
		});

		const wrapper = shallow(<GifGrid word={word} />);
		expect(wrapper).toMatchSnapshot();
	});

	test("It should show gifs when useFetchGifs is loaded", () => {
		const gifs = [
			{
				id: "123",
				url: "https://localhost/any/gif.gif.",
				title: "A cool gif",
				user: "userName",
				avatar: "https://localhost/any/avatar.png."
			},
			{
				id: "456",
				url: "https://localhost/any/gif.gif.",
				title: "A cool gif",
				user: "userName",
				avatar: "https://localhost/any/avatar.png."
			}
		];

		useFetchGifs.mockReturnValue({
			loading: false,
			gifs: gifs
		});

		const wrapper = shallow(<GifGrid word={word} />);
		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find("p").exists()).toBe(false);
		expect(wrapper.find("GifCard").length).toBe(gifs.length);
	});
});
