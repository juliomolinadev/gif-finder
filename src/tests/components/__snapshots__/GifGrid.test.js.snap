// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <GifGrid /> Component should be displayed correctly 1`] = `
<Fragment>
  <div
    className="d-flex justify-content-center mt-5"
  >
    <h3
      className=""
    >
      Gifs for "
      hi
      "
    </h3>
    <p>
      Loadian ...
    </p>
  </div>
  <div
    className="row row-cols-1 row-cols-md-2 row-cols-lg-3 g-3 mt-5"
  />
</Fragment>
`;

exports[`Tests in <GifGrid /> It should show gifs when useFetchGifs is loaded 1`] = `
<Fragment>
  <div
    className="d-flex justify-content-center mt-5"
  >
    <h3
      className=""
    >
      Gifs for "
      hi
      "
    </h3>
  </div>
  <div
    className="row row-cols-1 row-cols-md-2 row-cols-lg-3 g-3 mt-5"
  >
    <div
      className="d-flex justify-content-center col"
      key="123"
    >
      <GifCard
        gif={
          Object {
            "avatar": "https://localhost/any/avatar.png.",
            "id": "123",
            "title": "A cool gif",
            "url": "https://localhost/any/gif.gif.",
            "user": "userName",
          }
        }
      />
    </div>
    <div
      className="d-flex justify-content-center col"
      key="456"
    >
      <GifCard
        gif={
          Object {
            "avatar": "https://localhost/any/avatar.png.",
            "id": "456",
            "title": "A cool gif",
            "url": "https://localhost/any/gif.gif.",
            "user": "userName",
          }
        }
      />
    </div>
  </div>
</Fragment>
`;
